lightning-graph-sync/src/processing.rs

```diff
@@ -0,0 +1,473 @@
+use std::io;
+use std::io::Read;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use bitcoin::BlockHash;
+
+use lightning::ln::msgs::{
+	DecodeError, ErrorAction, OptionalField, UnsignedChannelAnnouncement, UnsignedChannelUpdate,
+};
+use lightning::routing::network_graph;
+use lightning::routing::network_graph::ChannelMutationProperties;
+use lightning::util::ser::Readable;
+
+use crate::error::GraphSyncError;
+
+pub fn update_network_graph(
+	network_graph: &network_graph::NetworkGraph,
+	update_data: &[u8],
+) -> Result<(), GraphSyncError> {
+	let mut read_cursor = io::Cursor::new(update_data);
+	read_network_graph(&network_graph, &mut read_cursor)
+}
+
+pub(crate) fn read_network_graph<R: Read>(
+	network_graph: &network_graph::NetworkGraph,
+	read_cursor: &mut R,
+) -> Result<(), GraphSyncError> {
+	let mut prefix = Vec::with_capacity(4);
+	prefix.resize(4, 0);
+	read_cursor.read_exact(&mut prefix)?;
+
+	let _deserialize_unsigned = match &prefix[..] {
+		&[76, 68, 75, 2] => true,
+		_ => {
+			return Err(GraphSyncError::ProcessingError(
+				"Prefix must equal 'LDK' and the byte 2".to_string(),
+			));
+		}
+	};
+
+	let chain_source: Option<Box<dyn lightning::chain::Access>> = None;
+	let current_timestamp = SystemTime::now()
```

Comment:
We're gonna get this from the update file itself, right?

Reply:
Yeah, I was actually debating as to whether to include the timestamp in the update data. I'll modify the serialization accordingly.

Reply:
Will update and reinitialize the server to have received timestamp fields tonight, and then regenerate test vectors.

Reply:
No.

Reply:
Hopefully is now, though I wanna refrain from pushing until my fuzz test's working.